zenoh-plugin-mqtt/src/lib.rs

```diff
@@ -444,12 +531,30 @@ async fn handshake_v3<'a>(
     handshake: v3::Handshake,
     zsession: Arc<Session>,
     config: Arc<Config>,
+    auth_dictionary: Arc<Option<HashMap<User, Password>>>,
 ) -> Result<v3::HandshakeAck<MqttSessionState<'a>>, MqttPluginError> {
     let client_id = handshake.packet().client_id.to_string();
-    log::info!("MQTT client {} connects using v3", client_id);
 
-    let session = MqttSessionState::new(client_id, zsession, config, handshake.sink().into());
-    Ok(handshake.ack(session, false))
+    match is_authorized(
+        (*auth_dictionary).as_ref(),
+        handshake.packet().username.as_ref(),
+        handshake.packet().password.as_ref(),
+    ) {
+        Ok(_) => {
+            log::info!("MQTT client {} connects using v3", client_id);
+            let session =
+                MqttSessionState::new(client_id, zsession, config, handshake.sink().into());
+            Ok(handshake.ack(session, false))
+        }
+        Err(err) => {
+            log::warn!(
+                "MQTT client {} connect using v3 rejected: {}",
+                client_id,
+                err
+            );
+            Ok(handshake.bad_username_or_pwd())
```

Comment:
`bad_username_or_pwd()` is for the case where the data in the user name or password is malformed.
See https://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718035 table 3.2.2.3.
I think `not_authorized()` is more appropriate here.

Reply:
Changed to `not_authorized()`.